internal/consensus/state_test.go

```diff
@@ -1661,26 +2183,30 @@ func TestCommitFromPreviousRound(t *testing.T) {
 	proposalCh := subscribe(ctx, t, cs1.eventBus, types.EventQueryCompleteProposal)
 
 	prop, propBlock := decideProposal(ctx, t, cs1, vs2, vs2.Height, vs2.Round)
-	propBlockHash := propBlock.Hash()
-	propBlockParts, err := propBlock.MakePartSet(partSize)
+	partSet, err := propBlock.MakePartSet(partSize)
 	require.NoError(t, err)
+	blockID := types.BlockID{
+		Hash:          propBlock.Hash(),
+		PartSetHeader: partSet.Header(),
+	}
 
 	// start round in which PO is not proposer
 	startTestRound(ctx, cs1, height, round)
 	ensureNewRound(t, newRoundCh, height, round)
 
 	// vs2, vs3 and vs4 send precommit for propBlock for the previous round
-	signAddVotes(ctx, t, config, cs1, tmproto.PrecommitType, propBlockHash, propBlockParts.Header(), vs2, vs3, vs4)
+	signAddVotes(ctx, t, cs1, tmproto.PrecommitType, config.ChainID(), blockID, vs2, vs3, vs4)
 
 	ensureNewValidBlock(t, validBlockCh, height, round)
 
 	rs := cs1.GetRoundState()
 	assert.True(t, rs.Step == cstypes.RoundStepCommit)
 	assert.True(t, rs.CommitRound == vs2.Round)
 	assert.True(t, rs.ProposalBlock == nil)
-	assert.True(t, rs.ProposalBlockParts.Header().Equals(propBlockParts.Header()))
-
-	if err := cs1.SetProposalAndBlock(ctx, prop, propBlock, propBlockParts, "some peer"); err != nil {
+	assert.True(t, rs.ProposalBlockParts.Header().Equals(blockID.PartSetHeader))
+	partSet, err = propBlock.MakePartSet(partSize)
+	require.NoError(t, err)
+	if err := cs1.SetProposalAndBlock(ctx, prop, propBlock, partSet, "some peer"); err != nil {
 		t.Fatal(err)
```

Comment:
we seem to be mixed on the idea of using t.Fatal vs require.NoError

Reply:
Fixed!